ansible/roles/test/files/ptftests/py3/hash_test.py

```diff
@@ -380,129 +377,209 @@ def check_ipv4_route(self, hash_key, src_port, dst_port_lists):
                             format(ip_src, ip_dst, src_port, rcvd_port, exp_src_mac, actual_src_mac))
         return (rcvd_port, rcvd_pkt)
 
-    def check_ipv6_route(self, hash_key, src_port, dst_port_lists):
+    def apply_mask_to_exp_pkt(self, masked_exp_pkt, version='IP'):
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        # mask the chksum also if masking the ttl
+        if self.ignore_ttl:
+            if version == 'IP':
+                masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "ttl")
+                masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+            else:
+                masked_exp_pkt.set_do_not_care_scapy(scapy.IPv6, "hlim")
+            masked_exp_pkt.set_do_not_care_scapy(scapy.TCP, "chksum")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        return masked_exp_pkt
+
+    def check_ipv4_route(self, hash_key, src_port, dst_port_lists, outer_sport=None, outer_dst_ip=None,
```

Comment:
I don't suggest you merge the logic of all tests into one check_ipv4_route and check_ipv6_route. It's hard to understand. If you want to reduce the duplication, you can have the base check_ipv4_route and check_ipv6_route in the base class and inherit them and override them in the sub classes.
And I strongly suggest you separate this change to another PR and only add the retry for all cases in this PR. For now the change is too big and hard to review.